### PR TITLE
Logging for requests and responses in retry section

### PR DESCRIPTION
**Pull request by ameer50**
# Which issue does this PR close?

Adding logs for request and response from object store to diagnose issues with occasional silently failing copies

# What changes are included in this PR?

Additional logging for debugging issues related to failed copies

# Are there any user-facing changes?

No

### CODE REVIEW

File: object_store/src/client/retry.rs [hunk 1]
```diff
@@ -190,70 +190,87 @@ impl RetryExt for reqwest::RequestBuilder {
...
+                                    true => Err(Error::BareRedirect),
+                                    false => Err(Error::Client {
+                                        body: None,
+                                        status: r.status(),
+                                    }),
+                                };
+                            }
+                            Err(e) => {
+                                // Log the error message
+                                error!("Error in response: {}", e);
```
Comment (asura-io) [inline object_store/src/client/retry.rs]: For ease of reading, do you think we should add both `e` and `r` in the log message?

File: object_store/src/client/retry.rs [hunk 2]
```diff
@@ -190,70 +190,87 @@ impl RetryExt for reqwest::RequestBuilder {
...
+                                {
+                                    let response_body = match r.text().await {
                                         Ok(body) => {
-                                            Error::Client {
-                                                body: Some(body).filter(|b| !b.is_empty()),
-                                                status,
-                                            }
-                                        }
+                                            let filtered_body = Some(body).filter(|b| !b.is_empty());
+                                            info!("NEW: Filtered response body: {:?}", filtered_body);
```
Comment (asura-io) [inline object_store/src/client/retry.rs]: Are we sure the `RUST_LOG_LEVEL` is set on our prod instance to get `info` logs?

File: object_store/src/client/retry.rs (hunk 1 above)
Comment (ameer50) [reply object_store/src/client/retry.rs]: Yeah good call, let me update that

File: object_store/src/client/retry.rs (hunk 2 above)
Comment (ameer50) [reply object_store/src/client/retry.rs]: Yep, the info level log we currently see in production as part of retry.s is the 

`"Encountered server error, backing off for...`

but more importantly, we se the RUST_LOG level to debug so INFO level is certainly captured

**Pull request closed without merging**